### PR TITLE
Additional settings shared component

### PR DESCRIPTION
**Pull request by FarajiA**
## JIRA:
See ticket here - https://jira.rax.io/browse/MNRVA-470

 ## Description:
This component is intended for use when updating or creating a monitor. The user needs a way to edit the additional settings portion of a monitor, this component will be used across two different views.

 ## Testing:
`npm run test`

 ## Screenshots:
addSettingsUpdate

### CODE REVIEW

**Comment by FarajiA** [comment]
I've gotten this in a working place, there's one test I'm having trouble with that uses a child component.

**Comment by FarajiA** [comment]
Issues appearing:

1. If monitor is saved with ResourceID when loading form excluded is disabled
2. If ResourceID is saved when re-editing in the same view the ResourceID dropdown doesn't

File: src/app/_features/monitors/components/additional-settings/additional-settings.component.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    if (formExport.resourceId) {
+      delete formExport.excludedResourceIds
+    }
+    else {
+      delete formExport.resourceId;
+      let excluded = [];
+      formExport.excludedResourceIds.map((item) => {
+        console.log('**excludedResourceId Object: ', item);
+        if (item['resource'] != "") {
+          console.log("**Adding resource: ", item['resource']);
```
Comment (lovedeep92) [inline src/app/_features/monitors/components/additional-settings/additional-settings.component.ts]: is this console necessary?

File: src/app/_features/monitors/components/additional-settings/additional-settings.component.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+          excluded.push(item['resource']);
+        }
+      });
+      formExport[CntrlAttribute.excludedResourceIds] = excluded;
+    }
+
+    return formExport;
+
+    /*
+ Object.keys(addFormValue).map((value) => {
```
Comment (lovedeep92) [inline src/app/_features/monitors/components/additional-settings/additional-settings.component.ts]: I think we can remove if not in use!

File: src/app/_features/monitors/components/additional-settings/additional-settings.component.ts (hunk 1 above)
Comment (FarajiA) [reply src/app/_features/monitors/components/additional-settings/additional-settings.component.ts]: Nope not needed, this was removed in the latest update.

File: src/app/_features/monitors/components/additional-settings/additional-settings.component.ts (hunk 2 above)
Comment (FarajiA) [reply src/app/_features/monitors/components/additional-settings/additional-settings.component.ts]: Yup, it's no longer present.

**Comment by lovedeep92** [comment]
Looks fine .. good to go 👍

File: src/app/_features/monitors/pages/create/monitor-create.page.ts [hunk 3]
```diff
@@ -51,26 +51,24 @@ change = false;
...
+    private fb: FormBuilder,
+    private labelService: LabelService, 
+    private router: Router, 
+    private readonly schemaService: SchemaService,
+    private resourceService: ResourcesService, 
+    private logService: LoggingService,
+    private cd: ChangeDetectorRef) {
+      
     }
+    ngAfterViewInit() {
```
Comment (pras7430) [inline src/app/_features/monitors/pages/create/monitor-create.page.ts:69]: Added due to below error on create monitor page.
expression has changed after it was checked. previous value 'null'. current value 'undefined'

File: src/app/_features/monitors/pages/details/monitor-details.page.spec.ts [hunk 4]
```diff
@@ -216,6 +221,21 @@ describe('MonitorDetailComponent', () => {
     expect(spyCompMethod).toHaveBeenCalled();
   });
 
+  it('should update Monitor settings', () => {
```
Comment (pras7430) [inline src/app/_features/monitors/pages/details/monitor-details.page.spec.ts:224]: I have removed it, as its entire definition is commented

**Pull request merged**